app/models/user.rb

```diff
@@ -3,12 +3,40 @@ class User < ActiveRecord::Base
   validates :password, length: {minimum: 5, maximum: 30}
   validates_uniqueness_of :email_address 
   has_many :reviews
-  has_many :queue_items
+  has_many :queue_items, -> { order(:order) }
   has_secure_password validations: false
 
+  def most_recent_review(video_id)
+    Review.where(user_id: self.id, video_id: video_id).order('created_at DESC').first
+  end
+
   def has_queue_item?(queue_id)
     QueueItem.find(queue_id).user == self
   end
 
+  def normalize_queue_item_order 
+    queue_items.each_with_index do |queue_item, index|
+      queue_item.update_attributes(order: index+1)
+    end
+  end
  
+  def update_queue_items(queue_items_data)
```

Comment:
Having this in `User` model is actually not a good practice due to tied dependency with params. Check out solution [here](https://www.gotealeaf.com/lessons/00bcc3b5/assignments/1569) - first minute should be it.